hoomd/md/pair/aniso.py

```diff
@@ -0,0 +1,222 @@
+    separation :math:`\vec r` and on the orientations :math:`\vec q_i`,
+    :math:`q_j`, of the particles.
+
+    `AnisotropicPair` is similiar to `Pair` except it does not support the
+    `xplor` shifting mode or `r_on`.
+
+    Args:
+        nlist (hoomd.md.nlist.NList) : The neighbor list.
+        r_cut (`float`, optional) : The default cutoff for the potential,
+            defaults to ``None`` which means no cutoff.
+        mode (`str`, optional) : the energy shifting mode, defaults to "none".
+    """
+
+    def __init__(self, nlist, r_cut=None, mode="none"):
+        self._nlist = OnlyType(md.nlist.NList, strict=True)(nlist)
+        tp_r_cut = TypeParameter('r_cut', 'particle_types',
+                                 TypeParameterDict(positive_real, len_keys=2)
+                                 )
+        if r_cut is not None:
+            tp_r_cut.default = r_cut
+        self._param_dict.update(
+            ParameterDict(mode=OnlyFrom(['none', 'shift'])))
+        self.mode = mode
+        self._add_typeparam(tp_r_cut)
+
+    def _return_type_shapes(self):
+        type_shapes = self.cpp_force.getTypeShapesPy()
+        ret = [json.loads(json_string) for json_string in type_shapes]
+        return ret
+
+
+class Dipole(AnisotropicPair):
+    R""" Screened dipole-dipole interactions.
+
+    Args:
+        nlist (`hoomd.md.nlist.NList`): Neighbor list
+        r_cut (float): Default cutoff radius (in distance units).
+        r_on (float): Default turn-on radius (in distance units).
+        mode (str): energy shifting/smoothing mode
+
+    `Dipole` computes the (screened) interaction between pairs of
+    particles with dipoles and electrostatic charges. The total energy
+    computed is:
+
+    .. math::
+
+        U_{dipole} = U_{dd} + U_{de} + U_{ee}
+
+        U_{dd} = A e^{-\kappa r}
+            \left(\frac{\vec{\mu_i}\cdot\vec{\mu_j}}{r^3}
+                  - 3\frac{(\vec{\mu_i}\cdot \vec{r_{ji}})
+                           (\vec{\mu_j}\cdot \vec{r_{ji}})}
+                          {r^5}
+            \right)
+
+        U_{de} = A e^{-\kappa r}
+            \left(\frac{(\vec{\mu_j}\cdot \vec{r_{ji}})q_i}{r^3}
+                - \frac{(\vec{\mu_i}\cdot \vec{r_{ji}})q_j}{r^3}
+            \right)
+
+        U_{ee} = A e^{-\kappa r} \frac{q_i q_j}{r}
+
+    See `Pair` for details on how forces are calculated and the
+    available energy shifting and smoothing modes.  Use ``params`` dictionary to
+    set potential coefficients. The coefficients must be set per unique pair of
+    particle types.
+
+    Attributes:
+        params (TypeParameter[tuple[``particle_type``, ``particle_type``], dict]):
+            The dipole potential parameters. The dictionary has the following
+            keys:
+
+            * ``A`` (`float`, **optional**) - :math:`A` - electrostatic energy
+              scale (*default*: 1.0)
+
+            * ``mu`` (`float`, **required**) - :math:`\mu` - emagnitude of
```

Comment:
`mu` is not a key in the `params` TypeParameter, it is its own type parameter and accepts a full vector.